### PR TITLE
Remove check in `check_error!`

### LINKED ISSUE #669: `check_error!` returns `Success` but doesn't set it in `sol.retcode`
**Describe the bug 🐞**

If I run an integrator to the end with `step!` and call `check_error` I get `ReturnCode.Success` (`sol.retcode` is `ReturnCode.Default`). The docstring for `check_error!` states:

> Same as `check_error` but also set solution's return code

However it doesn't set the solution's return code, but leaves it on `ReturnCode.Default`. I see `solve!` [handles this case specifically](https://github.com/SciML/OrdinaryDiffEq.jl/blob/c04c54bac08405100c2433ac4c172916dc594e28/src/solve.jl#L561-L564) and therefore doesn't see this issue.

Not sure what the best fix is. The return code is not set due to this if condition being false:

https://github.com/SciML/SciMLBase.jl/blob/d947bed3aa31cdf54791930cd8272306d6bd0f20/src/integrator_interface.jl#L646

Should that condition perhaps be removed? Or are integration interface users expected to convert Default to Success themselves?

**Minimal Reproducible Example 👇**

```julia

using OrdinaryDiffEq, SciMLBase

f(u, p, t) = 1.01 * u
u0 = 1 / 2
tspan = (0.0, 1.0)
prob = ODEProblem(f, u0, tspan)
integrator = init(prob, Tsit5(); reltol = 1e-8, abstol = 1e-8)
step!(integrator, tspan[end])

integrator.sol.retcode  # Default
check_error(integrator)  # Success
SciMLBase.check_error!(integrator) # Success
integrator.sol.retcode  # still Default
```

### PR DESCRIPTION
**Pull request by visr**
Fixes #669.

I'm not sure why this check was there. Was it just to save work? Why should the postamble not run if it is successful?

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
Does this allocation get elided?

**Comment by visr** [comment]
Running this on the example from #669 I don't see any allocations, nor do I see it in a profile.

```
@btime SciMLBase.check_error!($integrator);
9.510 ns (0 allocations: 0 bytes)
```

I decided to give AllocCheck a try. It just triggered this which looks unrelated:

```
using AllocCheck
@check_allocs g(i, code) = SciMLBase.solution_new_retcode(i.sol, code)
g(integrator, ReturnCode.Default)

ERROR: @check_allocs function encountered 1 errors (1 allocations / 0 dynamic dispatches).

Allocating runtime call to "jl_get_pgcstack" in unknown location
```

Running AllocCheck on `check_error!` does give some allocations from `postamble!` calling `resize!(integrator.sol.t, integrator.saveiter)`.

**Comment by ChrisRackauckas** [comment]
I mean in the context of a normal solve. Elision should happen in isolation but I want to make sure the compiler doesn't hit some limit say on Lorenz.

**Comment by visr** [comment]
I saw allocations coming from `postamble!`. Since that isn't what I'm interested in, I put that back in the if block.

Now the number of allocations for Lorentz is all from setup, just like in this example:

https://docs.sciml.ai/DiffEqDocs/stable/tutorials/faster_ode_example/#Example-Accelerating-a-Non-Stiff-Equation:-The-Lorenz-Equation

File: test/integrator_tests.jl [hunk 1]
```diff
@@ -62,3 +67,9 @@ for (uprev, tprev, u, t) in intervals(DummyIntegrator())
 end
 @test eltype(collect(intervals(DummyIntegrator()))) ==
       Tuple{Vector{Float64}, Float64, Vector{Float64}, Float64}
+
+@test integrator.sol.retcode == ReturnCode.Default
+@test check_error(integrator) == ReturnCode.Success
+@test integrator.sol.retcode == ReturnCode.Default
+@test SciMLBase.check_error!(integrator) == ReturnCode.Success
+@test integrator.sol.retcode == ReturnCode.Success
```
Comment (ChrisRackauckas) [inline test/integrator_tests.jl:75]: Add a test that nothing is allocated and this is good to go.

**Comment by ChrisRackauckas** [comment]
Sorry for the delayed review, but add an allocs test and this is good.

File: test/integrator_tests.jl (hunk 1 above)
Comment (visr) [reply test/integrator_tests.jl:76]: Is 14f6bced0e77b3075713d56cd73d4f56448c46b4 ok? Or do you also want to test this on Lorentz using OrdinaryDiffEq?

**Pull request merged**